### PR TITLE
Eta-expand primitive operators on bare reference (eigentrust pitfall #1)

### PR DESCRIPTION
**Pull request by kumavis**
Fixes pitfall **#1** from `docs/tracking/2026-04-23_eigentrust_pitfalls.md`.

## Summary

Primitive operators like `int+`, `rat*`, `rat-abs` are preparser keywords backed by `expr-int-add` / `expr-rat-mul` / `expr-rat-abs` AST structs, not regular functions. The user-facing reproducer:

```prologos
[map rat-abs xs]               ;; Unbound variable rat-abs
[zip-with rat+ xs ys]          ;; Multiplicity violation
```

The fix: in `elaborate-var`'s final fallback (just before the "Unbound variable" error), detect bare references to primitive operator names and eta-expand them into a concrete `expr-lam` whose body re-applies the corresponding AST constructor. This mirrors the pre-existing `'suc` pattern at the same call site. The parser keyword fast-path (`[int+ 3 4]` → `surf-int-add` directly) is untouched, so direct-application performance is unchanged.

## Asymmetry hint debunked

The pitfall report claimed `[foldr int+ ...]` worked while `[map rat-abs ...]` did not. Probing main showed both fail identically pre-fix with `Unbound variable`; both work post-fix. No special-case existed for `foldr` — the original report's "asymmetry" was likely a different case the user mis-attributed.

## Edge case answers

- `def f := int+` — works. `f : Int → Int → Int`. Tested via `[my-add 10 20] = 30`.
- `'[int+ int*]` — partial: with explicit `[List <Int → Int → Int>]` annotation it works; without one, list-element-type inference for bare function values is an orthogonal limitation (not in this pitfall's scope).

## Test plan

- [x] New `tests/test-prim-op-firstclass.rkt` — 18 cases covering the eigentrust reproducer, multiple primitive families (int+, int*, rat+, rat*, rat-abs, rat-lt, etc.), variable binding, in-application-position regression, list-of-functions edge cases
- [x] Regression checks: `test-int` (60 tests), `test-rat`, `test-elaborator` (40 tests), `test-parse-integration` (47 tests) — all green
- [x] CI fix included (`benchmarks/micro/info.rkt` `compile-omit-paths` for the stale `bench-bsp-le-track2.rkt`) so `raco pkg install` succeeds

## Commits

1. `adea436` — primary fix in `elaborator.rkt` + new test file
2. `2982bc8` — CI fix (skip stale bench file)

https://claude.ai/code/session_01MbncYJnrvjzhbVWw4xGi5x

---
_Generated by [Claude Code](https://claude.ai/code/session_01MbncYJnrvjzhbVWw4xGi5x)_

### CODE REVIEW

File: racket/prologos/elaborator.rkt [hunk 1]
```diff
@@ -749,8 +749,92 @@
...
+      ;; consumed by the parser. In value position (passed to map/zip-with/foldr,
+      ;; bound to a variable, stored in a list, etc.) the bare identifier
+      ;; produces surf-var here. Eta-expand to a lambda so the operator becomes
+      ;; first-class. Mirrors the `suc` pattern above.
+      [(primitive-op-eta-expansion name)
+       => (lambda (e) e)]
       [else (unbound-variable-error loc "Unbound variable" name)])))
 
+;; ========================================
+;; Eigentrust pitfall #1: primitive operators as first-class values
```
Comment (kumavis) [inline racket/prologos/elaborator.rkt]: remove references to the pitfalls doc in source (in tests is fine)

**Review by kumavis — COMMENTED** [review]
hiero approves

**Review by hierophantos — APPROVED** [review]
LGTM. Surgical extension to `elaborate-var`'s existing fallback: just before the "Unbound variable" error, check `primitive-op-eta-table` and produce an eta-expanded `expr-lam` if the name resolves. Mirrors the pre-existing `'suc` eta-expansion (one cond-arm above). Parser keyword fast-path (`[int+ 3 4]` → `surf-int-add` directly) untouched.

The `mw` multiplicity choice for the lambda parameters is the right QTT call — function values must be `mw` so closures capturing environment values don't trip linearity. Matches the `'suc` precedent.

Coverage: 24 entries spanning Int (`+ - * / mod neg abs lt le eq` + `from-nat`) and Rat (`+ - * / neg abs lt le eq` + `from-int rat-numer rat-denom`). Reasonable surface for the operators users pass to higher-order functions.

Test discipline:
- Direct-application regressions confirm parser keyword fast-path preserved
- Pitfall reproducers (`[map rat-abs ...]`, `[foldr int+ ...]`) pin the failure mode
- Variable-binding edge case (`def my-add := int+`) works as expected
- Honest about the orthogonal list-element-type-inference limitation

Mergeable: yes (clean, no conflicts). 0 existing codebase usage of prim ops in non-application position (was broken; fix is purely additive).

Approving.
